src/Components/Projects/Projects.tsx

```diff
@@ -164,6 +168,22 @@ const Projects: React.FC = () => {
       });
   };
   // *****************************************************
+  // **********search by proj name***********************
+  const getProjectTitleValue = (e: MouseEvent) => {
+    getAllProjectsList(1, e.target.value);
+    setSearchString(e.target.value);
+  };
+  // ************************
+
+  useEffect(() => {
```

Comment:
Debouncing is in place, but there's a small glitch: when checking the network tab, you'll notice a network request being triggered with every keystroke. 
Try figuring out what is going wrong.

![debouncing](https://github.com/Rawan-Mohamed/project-management-system-1/assets/64327685/f056798c-ac9c-4731-a40f-9f6660cb45df)

Reply:
If you encounter any challenges figuring out what is going wrong, feel free to reach out to @Rawan-Mohamed , as she has experience dealing with a similar issue.

Reply:
done